src/init/mod.rs

```diff
@@ -121,68 +121,50 @@ pub fn init_stub(shell_name: &str) -> io::Result<()> {
              * https://github.com/starship/starship/pull/241
              * https://github.com/starship/starship/pull/278
              */
-            let script = {
-                format!(
-                    r#"if [ "${{BASH_VERSINFO[0]}}" -gt 4 ] || ([ "${{BASH_VERSINFO[0]}}" -eq 4 ] && [ "${{BASH_VERSINFO[1]}}" -ge 1 ])
-then
-source <("{0}" init bash --print-full-init)
-else
-source /dev/stdin <<<"$("{0}" init bash --print-full-init)"
-fi"#,
-                    starship.sprint_posix()?
-                )
-            };
-
-            Some(script)
-        }
-        "zsh" => {
-            let script = format!(
-                "source <(\"{}\" init zsh --print-full-init)",
-                starship.sprint_posix()?
-            );
-            Some(script)
-        }
-        "fish" => {
+            r#"
+            major="${{BASH_VERSINFO[0]}}"
+            minor="${{BASH_VERSINFO[1]}}"
+
+            if ((major > 4)) || {{ ((major == 4)) && ((minor >= 1)); }}; then
+                source <("{0}" init bash --print-full-init)
+            else
+                source /dev/stdin <<<"$("{0}" init bash --print-full-init)"
+            fi
+            "#,
```

Comment:
I've tested the following snippet using versions `bash-3.2`, `bash-4.4`, and `bash-5.1`. It passes shellcheck, uses arithmetic comparisons, and avoids using subshell for the right `OR` branch (check for version `>=4.1`).
Note that Bash isn't indentation sensitive so indenting this doesn't really change anything except it doesn't stick out anymore.

Reply:
`IIFE`-like `__main` is used to allow sourcing this stub without polluting scope.